src/aind_metadata_service/sharepoint/client.py

```diff
@@ -366,64 +597,102 @@ def _handle_response_from_sharepoint(  # noqa: C901
             procedures = Procedures.construct(subject_id=subject_id)
-            head_frames = []
-            injections = []
-            fiber_implants = []
-            craniotomies = []
-            for list_item in list_items:
-                if list_item.get_property(list_fields.PROCEDURE.value):
-                    procedure_types = list_item.get_property(
-                        list_fields.PROCEDURE.value
-                    ).split(str_helpers.PROCEDURE_TYPE_SPLITTER.value)
-                else:
-                    procedure_types = []
-                for procedure_type in procedure_types:
-                    if procedure_type == nsb_proc_types.HEAD_PLANT.value:
-                        head_frames.append(
-                            self._map_list_item_to_head_frame(list_item)
-                        )
-                    if procedure_type in {
-                        nsb_proc_types.STEREOTAXIC_INJECTION.value,
-                        nsb_proc_types.INJECTION.value,
-                        nsb_proc_types.INJ.value,
-                    }:
-                        injections.append(
-                            self._map_list_item_to_injection(list_item)
-                        )
-                    if (
-                        procedure_type
-                        == nsb_proc_types.OPTIC_FIBER_IMPLANT.value
-                    ):
-                        fiber_implants.append(
-                            self._map_list_item_to_fiber_implant(list_item)
-                        )
-                    if procedure_type in {
-                        nsb_proc_types.WHOLE_HEMISPHERE_CRANIOTOMY_NP.value,
-                        nsb_proc_types.C_MULTISCOPE.value,
-                        nsb_proc_types.C_CAM.value,
-                        nsb_proc_types.C.value,
-                    }:
-                        craniotomies.append(
-                            self._map_list_item_to_craniotomy(list_item)
-                        )
-                if head_frames:
-                    procedures.headframes = head_frames
-                if injections:
-                    procedures.injections = injections
-                if fiber_implants:
-                    procedures.fiber_implants = fiber_implants
-                if craniotomies:
-                    procedures.craniotomies = craniotomies
+            if head_frames:
+                procedures.headframes = head_frames
+            if injections:
+                procedures.injections = injections
+            if fiber_implants:
+                procedures.fiber_implants = fiber_implants
+            if craniotomies:
+                procedures.craniotomies = craniotomies
             response = Responses.model_response(procedures)
         else:
             response = Responses.no_data_found_response()
         return response
 
+    def _map_2023_response(self, list_items: ListItemCollection):
```

Comment:
This might be a good opportunity to practice defining return types. Here and in the other functions, please specify the expected return type (the `-> ???` thing)